modules/apps/commerce/commerce-product-ddm-impl/src/main/java/com/liferay/commerce/product/ddm/internal/DDMHelperImpl.java

```diff
@@ -235,51 +234,49 @@ private String _createDDMFormRuleInputMapping(
 
 		List<DDMFormField> ddmFormFields = ddmForm.getDDMFormFields();
 
-		Stream<DDMFormField> stream = ddmFormFields.stream();
-
-		Stream<String> inputMappingStatementStream = stream.map(
-			field -> String.format(
-				"'%s=', getValue('%s')", field.getName(), field.getName()));
-
-		inputMappingStatementStream = Stream.concat(
-			Stream.of(String.format("'companyId=%s'", companyId)),
-			inputMappingStatementStream);
-
-		inputMappingStatementStream = Stream.concat(
-			Stream.of(String.format("'cpDefinitionId=%s'", cpDefinitionId)),
-			inputMappingStatementStream);
-
-		inputMappingStatementStream = Stream.concat(
-			Stream.of(String.format("'groupId=%s'", groupId)),
-			inputMappingStatementStream);
-
-		inputMappingStatementStream = Stream.concat(
-			Stream.of(
-				String.format("'commerceAccountId=%s'", commerceAccountId)),
-			inputMappingStatementStream);
-
-		inputMappingStatementStream = Stream.concat(
-			Stream.of(String.format("'userId=%s'", userId)),
-			inputMappingStatementStream);
-
-		inputMappingStatementStream = Stream.concat(
-			Stream.of(
-				String.format("'locale=%s'", LocaleUtil.toLanguageId(locale))),
-			inputMappingStatementStream);
+		StringBundler sb = new StringBundler((ddmFormFields.size() * 2) + 11);
+
+		sb.append(String.format("'companyId=%s'", companyId));
```

Comment:
@julschong Could you explore the possibility to avoid using `String.format` here, especially when used together with `StringBundler`? Let's see how complicated the code will be.